server/monitor/src/main/resources/resources/functions.js

```diff
@@ -0,0 +1,686 @@
+    if (rows.length > 2) {
+      tmpH.classList.add('sortable');
+    }
+    $(tmpH.getElementsByTagName('span')).remove();
+    count += 1;
+  }
+
+  // If there are more than 2 rows, add arrow to the selected column
+  if (rows.length <= 2) {
+      switching = false;
+  } else {
+    h = rows[0].getElementsByTagName('th')[n];
+    if (dir == 'asc') {
+      $(h).append('<span class="glyphicon glyphicon-chevron-up"' +
+          ' width="10px" height="10px" />');
+    } else if (dir == 'desc') {
+      $(h).append('<span class="glyphicon glyphicon-chevron-down"' +
+          ' width="10px" height="10px" />');
+    }
+  }
+
+  /*
+   * Make a loop that will continue until
+   * no switching has been done:
+   */
+  while (switching) {
+    switching = false;
+    rows = table.getElementsByTagName('tr');
+
+    /*
+     * Loop through all table rows (except the
+     * first, which contains table headers):
+     */
+    for (i = 1; i < (rows.length - 1); i++) {
+      shouldSwitch = false;
+      /*
+       * Get two elements to compare,
+       * one from current row and one from the next:
+       * If the element is a dash, convert to null, otherwise,
+       * if it is a string, convert to number
+       */
+      x = rows[i].getElementsByTagName('td')[n].getAttribute('data-value');
+      xFinal = (x === '-' || x === '&mdash;' ?
+          null : (Number(x) == x ? Number(x) : x));
+
+      y = rows[i + 1].getElementsByTagName('td')[n].getAttribute('data-value');
+      yFinal = (y === '-' || y === '&mdash;' ?
+          null : (Number(y) == y ? Number(y) : y));
+
+      /*
+       * Check if the two rows should switch place,
+       * based on the direction, asc or desc:
+       */
+      if (dir == 'asc') {
+        if (xFinal > yFinal || (xFinal !== null && yFinal === null)) {
+          // if so, mark as a switch and break the loop:
+          shouldSwitch = true;
+          break;
+        }
+      } else if (dir == 'desc') {
+        if (xFinal < yFinal || (yFinal !== null && xFinal === null)) {
+          // if so, mark as a switch and break the loop:
+          shouldSwitch = true;
+          break;
+        }
+      }
+    }
+    if (shouldSwitch) {
+      /*
+       * If a switch has been marked, make the switch
+       * and mark that a switch has been done:
+       */
+      rows[i].parentNode.insertBefore(rows[i + 1], rows[i]);
+      switching = true;
+    }
+  }
+}
+
+/**
+ * Clears the selected table while leaving the headers
+ *
+ * @param {string} tableID Table to clear
+ */
+function clearTable(tableID) {
+  // JQuery selector to select all rows except for the first row (header)
+  $('#' + tableID).find('tr:not(:first)').remove();
+}
+
+///// REST Calls /////////////
+
+/**
+ * REST GET call for the master information,
+ * stores it on a sessionStorage variable
+ */
+function getMaster() {
+  $.getJSON('/rest/master', function(data) {
+    sessionStorage.master = JSON.stringify(data);
```

Comment:
How does jquery react if it receives some non HTTP/200 response? (e.g. if there was an Exception due to a bug in the Monitor rest call). Do we get a nice message?